File: features/promo/tokopoints/src/main/java/com/tokopedia/tokopoints/view/adapter/CatalogBannerPagerAdapter.java
package com.tokopedia.tokopoints.view.adapter;

import android.content.Context;
import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.URLUtil;
import android.widget.ImageView;

import com.tokopedia.abstraction.common.utils.image.ImageHandler;
import com.tokopedia.applink.RouteManager;
import com.tokopedia.tokopoints.R;
import com.tokopedia.tokopoints.view.model.CatalogBanner;
import com.tokopedia.tokopoints.view.presenter.CatalogListingPresenter;

import java.util.List;

public class CatalogBannerPagerAdapter extends PagerAdapter {

    private List<CatalogBanner> mItems;
    private LayoutInflater mInflater;
    private CatalogListingPresenter mPresenter;

    public CatalogBannerPagerAdapter(Context context, List<CatalogBanner> items, CatalogListingPresenter presenter) {
        this.mItems = items;
        mInflater = LayoutInflater.from(context);
        mPresenter = presenter;
    }

    @Override
    public void destroyItem(View container, int position, Object object) {
        ((ViewPager) container).removeView((View) object);
    }

    @Override
    public int getCount() {
        return mItems.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view.equals(object);
    }

    @Override
    public Object instantiateItem(ViewGroup view, final int position) {
        ImageView banner = (ImageView) mInflater.inflate(R.layout.tp_item_catalog_banner, view, false);
        ImageHandler.loadImageFit2(banner.getContext(), banner, mItems.get(position).getImageUrl());
        banner.setOnClickListener(view1 -> {
            if (URLUtil.isValidUrl(mItems.get(position).getRedirectUrl())) {
                mPresenter.getView().openWebView(mItems.get(position).getRedirectUrl());
            } else {
                if (mPresenter.getView().getActivityContext() != null) {
                    RouteManager.route(mPresenter.getView().getActivityContext(), mItems.get(position).getRedirectUrl());
                }
            }
        });
        view.addView(banner);

        return banner;
    }
}

File: features/promo/tokopoints/src/main/java/com/tokopedia/tokopoints/view/cataloglisting/CatalogBannerPagerAdapter.kt
package com.tokopedia.tokopoints.view.cataloglisting

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.URLUtil
import android.widget.ImageView
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import com.tokopedia.abstraction.common.utils.image.ImageHandler
import com.tokopedia.applink.RouteManager
import com.tokopedia.tokopoints.R
import com.tokopedia.tokopoints.view.model.CatalogBanner

class CatalogBannerPagerAdapter(context: Context?, private val mItems: List<CatalogBanner>, view: CatalogListingContract.View) : PagerAdapter() {
    private val mInflater: LayoutInflater
    private val mView: CatalogListingContract.View
    override fun destroyItem(container: View, position: Int, `object`: Any) {
        (container as ViewPager).removeView(`object` as View)
    }

    override fun getCount(): Int {
        return mItems.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view == `object`
    }

    override fun instantiateItem(view: ViewGroup, position: Int): Any {
        val banner = mInflater.inflate(R.layout.tp_item_catalog_banner, view, false) as ImageView
        ImageHandler.loadImageFit2(banner.context, banner, mItems[position].imageUrl)
        banner.setOnClickListener {
            if (URLUtil.isValidUrl(mItems[position].redirectUrl)) {
                mView.openWebView(mItems[position].redirectUrl)
            } else {
                if (mView.activityContext != null) {
                    RouteManager.route(mView.activityContext, mItems[position].redirectUrl)
                }
            }
        }
        view.addView(banner)
        return banner
    }

    init {
        mInflater = LayoutInflater.from(context)
        mView = view
    }
}
